tests(e2e): catch operator logs during the tests run

sxd opened the pull request:
Since we included the catch all for the logs of all the pods and show that it works,
now we can do the same for the operator logs using the same functions and model.

We gather the operator pods by matching the labels and we split them by different
pod name, meaning that if the operator is destroyed and recreated we will have a
different log file for the new pod.

Closes #5453

sxd committed:
chore(tests): catch operator logs during the tests run

Since we included the catch all for the logs of all the pods and show that it works,
now we can do the same for the operator logs using the same functions and model.

We gather the operator pods by matching the labels and we split them by different
pod name, meaning that if the operator is destroyed and recreated we will have
a different log file for the new pod.

Also, some clean up of repeated tailing logs to avoid more queries to the API and
unnecesary logs requirements.

Closes #5453

Signed-off-by: Jonathan Gonzalez V. <jonathan.gonzalez@enterprisedb.com>

mnencia committed:
chore: review

Signed-off-by: Marco Nenciarini <marco.nenciarini@enterprisedb.com>

mnencia committed:
test: focus to be reverted

Signed-off-by: Marco Nenciarini <marco.nenciarini@enterprisedb.com>

mnencia commented:
/test d=push tl=4 l=local

mnencia committed:
ci: save the logs

Signed-off-by: Marco Nenciarini <marco.nenciarini@enterprisedb.com>

mnencia commented:
/test d=push tl=4 l=local

mnencia committed:
ci: update make clean target

Signed-off-by: Marco Nenciarini <marco.nenciarini@enterprisedb.com>

mnencia commented:
Manual run to use the workflow definition on the branch https://github.com/cloudnative-pg/cloudnative-pg/actions/runs/10812091268

mnencia committed:
Revert "test: focus to be reverted"

This reverts commit e71a9db5fdad47ad4e226263a014b7567438a4f7.

Signed-off-by: Marco Nenciarini <marco.nenciarini@enterprisedb.com>

mnencia committed:
ci: update gitignore

Signed-off-by: Marco Nenciarini <marco.nenciarini@enterprisedb.com>

jsilvela commented on tests/utils/namespace.go:
```diff
@@ -175,38 +51,23 @@ func (env TestingEnvironment) GetOperatorLogs(buf *bytes.Buffer) error {
 	return streamPodLog.Stream(env.Ctx, buf)
 }
 
-// DumpNamespaceOperatorLogs writes the operator logs related to a namespace into a writer
-// and also into a file
-func (env TestingEnvironment) DumpNamespaceOperatorLogs(namespace, testName string, output io.Writer) {
```
IMO this is a loss. This code was ensuring we saved the operator logs for the test's namespace only. And would also show on Ginkgo operator logs for this namespace that were Error or Warn.
Is this functionality captured in a different way in this PR?

jsilvela replied:
I have no problem with the general operator log tailing being converted to stern.
It's the namespace-filtered operator logs that I think were an improvement over a single file with all the operator logs.

sxd replied:
@jsilvela after months of having this I haven't found any one using the operator logs in the Ginkgo namespace, instead, is more noise than help in there because start to fill up the Ginkgo buffer and one start to miss the focus on the error, about the namespace captured log, that's right, we may improve that in the future but for now an operator log that you can grep with the namespace looks fair enough, this is a long process of improving the e2e tests, for sure if we found needed we can incorporate that functionality in the next few improvements but for now I think it's ok to keep this out, there's other PRs that I'll be opening to improve the e2e

jsilvela replied:
> after months of having this I haven't found any one using the operator logs in the Ginkgo namespace
not been months.
This was merged 20 days ago. https://github.com/cloudnative-pg/cloudnative-pg/pull/5215
As to "not found anyone" -- right, so, I did use it, it helped for an investigation, which is why I decided to make the functionality available.

> we may improve that in the future but for now an operator log that you can grep with the namespace looks fair enough
This is not about adding code, but about deleting less of it.

that said, @sxd , I'm not going to ask for changes, go ahead. But I'm also not giving approve.

sxd committed:
chore: linter and remove SternMultiTailer struct

Signed-off-by: Jonathan Gonzalez V. <jonathan.gonzalez@enterprisedb.com>

fcanovai committed:
test: decouple the stern multitailer

Multitailer now isn't tied to the environment it is running into,
that knowledge is now in the environment.

Also, make the multitailer safe for multiple calls.

Signed-off-by: Francesco Canovai <francesco.canovai@enterprisedb.com>
Signed-off-by: Jonathan Gonzalez V. <jonathan.gonzalez@enterprisedb.com>

sxd commented:
/tests depth=push test_level=1 feature_type=basic

sxd commented:
/test depth=push test_level=1 feature_type=basic

fcanovai committed:
test: fix panic when report is nil

Signed-off-by: Francesco Canovai <francesco.canovai@enterprisedb.com>

fcanovai commented:
/test depth=push test_level=1 feature_type=basic
